src/test-apps/happy/lib/WeaveDeviceManager.py

```diff
@@ -446,6 +460,29 @@ def testWdmClientDataSinkRefreshGetDataRefresh(testObject):
     testObject.closeWdmClient()
     print("testWdmClientDataSinkRefreshGetDataRefresh completes")
 
+def testWdmClientDataSinkRefreshDataWithEvents(testObject):
+    import time
+    testObject.createWdmClient()
+    # We need to set node Id to get everything works
+    testObject.setNodeId(-2)
+    # We needs a data sink anyway
+    localeSettingsTrait = testObject.newDataSink(20, 0, "/")
+    testObject.refreshData(withEvents=True)
+    testObject.getEvents()
+    # Sleep 5 second to wait for more events generated by mock-device
+    time.sleep(5)
+    testObject.refreshData(withEvents=True)
```

Comment:
In this test, since it is large event stream, which needs multiple refreshData call, you may need to add validation to check the second Event batch from first refreshData would not overlap with previous events from refreshData. 

Everytime you call refreshData(withEvents=True), the previous event cache in wdmClienet should be cleared, and new events would be loaded.